### ISSUE TITLE
[$1000] Report item stays hovered after we edited the message or cancelled the edit message

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#14080]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Choose edit option for any message
2. Try to edit and save
3. Now hover on other rows and observe the edited message row

## Expected Result:
After edit/cancel edit the message item should not be hovered

## Actual Result:
After edit/cancel edit the message item keeps hovered

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?

- [ ] MacOS / Chrome / Safari

**Version Number:** 1.2.50-4
**Reproducible in staging?:** y 
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/211101017-19a13a28-9b4d-4900-bc5a-b524a8e43474.mov

https://user-images.githubusercontent.com/43996225/211101106-c32d65ef-0a17-43de-9add-5b667d993286.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @Pujan92 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1672687285656309

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01bc65e21d9f4a414d</li>
        <li>Upwork Job ID: 1613204426013020160</li>
        <li>Last Price Increase: 2023-01-11</li>
    </ul>
</details>

**Comment by amyevans** [comment]
I also could not repro in Safari 16.1 (I did not test older versions).

**Comment by isabelastisser** [comment]
I'm also unable to reproduce it:

https://user-images.githubusercontent.com/31258516/211357732-6b7bc926-d161-447a-bfa0-9d921abc27d1.mp4

**Comment by isabelastisser** [comment]
I'm able to reproduce it now.

https://user-images.githubusercontent.com/31258516/211854820-c447e0ce-db30-4d98-b5f0-a7286bd133bd.mp4

**Comment by s77rt** [comment]
### Proposal
```diff
diff --git a/src/components/Hoverable/index.js b/src/components/Hoverable/index.js
index ef4bc2bf53..841696f9ef 100644
--- a/src/components/Hoverable/index.js
+++ b/src/components/Hoverable/index.js
@@ -107,6 +107,15 @@ class Hoverable extends Component {
                         onMouseLeave(el);
                     }
                 },
+                onBlur: (el) => {
+                    this.setIsHovered(false);
+
+                    // Call the original onBlur, if any
+                    const {onBlur} = this.props.children;
+                    if (_.isFunction(onBlur)) {
+                        onBlur(el);
+                    }
+                },
             });
         }
         return (
@@ -115,6 +124,7 @@ class Hoverable extends Component {
                 ref={el => this.wrapperView = el}
                 onMouseEnter={() => this.setIsHovered(true)}
                 onMouseLeave={() => this.setIsHovered(false)}
+                onBlur={() => this.setIsHovered(false)}
             >
                 { // If this.props.children is a function, call it to provide the hover state to the children.
                     _.isFunction(this.props.children)
```


### Details
This is similar to another issue where the hovered item stays hovered due to the missing event `onMouseLeave`.
We can compensate this missing event by relaying on the `onBlur` event.

Note: The `resetHoverStateOnOutsideClick` can be removed since `onBlur` seems to do it's job, this it out of the scope of this issue, just worth noting. @roryabraham

**Comment by s77rt** [comment]
@roryabraham I have only tested with closing the modal (as described in the comments) is there any other scenarios that we should test before removing that code?

**Comment by roryabraham** [comment]
I can't think of any others, no

**Comment by isabelastisser** [comment]
Not overdue.

**Comment by ctkochan22** [comment]
The proposal looks good.

**Comment by mananjadhav** [comment]
I missed this but the proposal looks promising. 

I have one concern I haven't been able to reproduce this even once on Safari/Chrome on MacOS. @isabelastisser @kavimuru does it require any preconditions or any other pattern to reproduce this reliably?

@ctkochan22 Were you able to reproduce this?

**Comment by ctkochan22** [comment]
Yeah, its super tricky and very annoying.

It seems like you need a full page of comments. And you need to scroll up a bit, so you're not seeing the very last message.

The key, I think, is to click "Save changes", but some how have your mouse not hovering over the edited comment. I usually have to do it once or twice before getting it to work

**Comment by isabelastisser** [comment]
Update: 

- Hired @s77rt in Upwork for fixing the issue.
- Hired @mananjadhav in Upwork for reviewing the proposal (C+).

**Comment by Pujan92** [comment]
@isabelastisser Can you plz consider me as a Reporting of the issue

**Comment by isabelastisser** [comment]
Hey @Pujan92, sorry I missed the information above. I will hire you on Upwork now.

**Comment by s77rt** [comment]
Applied in [Upwork](https://www.upwork.com/jobs/~01bc65e21d9f4a414d).
[PR](https://github.com/Expensify/App/pull/14377) is ready.

**Comment by mananjadhav** [comment]
PR reviewed. Was able to reproduce the issue. 

> It seems like you need a full page of comments. And you need to scroll up a bit, so you're not seeing the very last message.

I think this was crucial for me to reproduce.

**Comment by getusha** [comment]
I believe the PR introduced a regression here https://expensify.slack.com/archives/C049HHMV9SM/p1674546335549909

**Comment by getusha** [comment]
https://user-images.githubusercontent.com/75031127/214243644-a3d1e5eb-3453-4a90-8892-f64a06116438.mov

This is what it looks on safari

**Comment by s77rt** [comment]
Thanks @getusha I didn't catch that while testing
The fix is coming up...

**Comment by s77rt** [comment]
The [PR](https://github.com/Expensify/App/pull/14507) fixing the regression is ready

**Comment by ctkochan22** [comment]
We are good here?

**Comment by isabelastisser** [comment]
Update: All contributors were paid in Upwork now. 

Reported by @Pujan92
Fixed by @s77rt 
C+ review @mananjadhav

**Comment by gadhiyamanan** [comment]
> The [PR](https://github.com/Expensify/App/pull/14507) fixing the regression is ready

I think this is eligible for reporting bonus as i found the regression
Cc: @isabelastisser @mananjadhav @ctkochan22

**Comment by gadhiyamanan** [comment]
Bump @isabelastisser @ctkochan22 @mananjadhav ^

**Comment by mananjadhav** [comment]
@isabelastisser @gadhiyamanan is eligible for the reporting payout here.

**Comment by ctkochan22** [comment]
I'm not 100% sure how this works if its a regression, but probably that is correct. @isabelastisser can we ask in slack?

**Comment by isabelastisser** [comment]
Hi @gadhiyamanan , I invited you to the job in Upwork now. Please accept it and I will process the payment. Thanks!

**Comment by gadhiyamanan** [comment]
@isabelastisser applied

**Comment by isabelastisser** [comment]
The payment was processed in Upwork, we are all set!

**Comment by gadhiyamanan** [comment]
@isabelastisser please close the upwork contract

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.81-1 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/14507


If no regressions arise, payment will be issued on 2023-03-17. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@mananjadhav / @s77rt] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@mananjadhav / @s77rt] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@mananjadhav / @s77rt] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@isabelastisser] Determine if we should create a regression test for the bug.
- [ ] [@mananjadhav / @s77rt] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to the [appropriate location](https://github.com/Expensify/App/blob/main/contributingGuides/TestRail_Navigation_C%2B.md) to ensure the same bug will not reach production again.
- [ ] [@isabelastisser] Review the proposed regression test steps and location.
- [ ] [@isabelastisser] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by MelvinBot** [comment]
Looks like something related to `react-navigation` may have been mentioned in this issue discussion.

As a reminder, please make sure that all proposals are not workarounds and that any and all attempt to fix the issue holistically have been made before proceeding with a solution. Proposals to change our `DeprecatedCustomActions.js` files should not be accepted.

Feel free to drop a note in [#expensify-open-source](https://expensify.slack.com/archives/C01GTK53T8Q) with any questions.

**Comment by s77rt** [comment]
- The PR that introduced the bug has been identified: I think this existed from first implementation cc @mananjadhav 
- The offending PR has been commented on: ^
- A discussion in #expensify-bugs has been started: cc @mananjadhav 

___

**Regression Test Proposal**

1. Open context menu for any message item
2. Choose edit comment
3. Click cancel or save changes
4. Verify that the message item is not hovered
5. Open an attachment (modal) in chat
6. Close the modal
7. Verify that the message item of that attachment is not hovered
